tc/sgx/trusted_worker_manager/enclave/kme/workload/kme_workload_plug-in.cpp

```diff
@@ -57,6 +75,268 @@ WPEInfo::WPEInfo(const ByteArray& _sk) {
+    if (pub_sk.VerifySignature(ByteArray(std::begin(concat_str_hash),
+                                         std::end(concat_str_hash)),
+                               uid_nonce_sig) != 1) {
+        this->SetStatus((int)ERR_KME_REPL_SIG_VERIF_FAILED, out_work_order_data);
+        ThrowIf<ValueError>(true, "Signature verification failed while getting state.");
+    }
+    //Generate Symmetric enc key
+    ByteArray sym_key = tcf::crypto::skenc::GenerateKey();
+
+    EnclaveData* enclave_data = EnclaveData::getInstance();
+    //Obtain state data. String concatenation of pvt_data, enc_key_map, sig_key_map
+    //delimited by semicolon (;)
+    std::string state_data = enclave_data->get_private_data();
+    state_data += ";";
+    state_data += KMEWorkloadProcessor::serializeEncKeyMap();
+    state_data += ";";
+    state_data += KMEWorkloadProcessor::serializeSigKeyMap();
+
+    ByteArray state_data_ba = StrToByteArray(state_data);
+    //Encrypt state data with sym enc key
+    ByteArray enc_state_data = tcf::crypto::skenc::EncryptMessage(
+                                   sym_key, state_data_ba);
+    //Encrypt sym enc key with requester pub enc key
+    tcf::crypto::pkenc::PublicKey pub_ek = tcf::crypto::pkenc::PublicKey(ByteArrayToStr(pub_enc_key_ba));
+    ByteArray enc_sym_key = pub_ek.EncryptMessage(sym_key);
+
+    //Generate signature of the output
+
+    //SHA256 of state data(JSON as byte array) is calculated
+    ComputeSHA256Hash(ByteArrayToString(state_data_ba), concat_str_hash);
+
+    //Concatenate hash, sym enc key, unique id, nonce
+    concat_ba = StrToByteArray( ByteArrayToStr(ByteArray(std::begin(concat_str_hash),
+                                               std::end(concat_str_hash)))
+                                + ByteArrayToStr(sym_key)
+                                + ByteArrayToStr(unique_id_hex)
+                                + ByteArrayToStr(generated_nonce_hex));
+
+    //SHA256 of concatenated byte array is calculated
+    ComputeSHA256Hash(ByteArrayToString(concat_ba), concat_str_hash);
+
+    //Sign hash with signing key
+    ByteArray hash_signed = enclave_data->sign_message(
+        ByteArray(std::begin(concat_str_hash), std::end(concat_str_hash)));
+    SetStatus(KME_REPL_OP_SUCCESS, out_work_order_data);
+    //@TODO - Output at index 1 to be Attestation info of this(primary) KME
+    AddOutput(1, unique_id_hex, out_work_order_data); // unique_id_hex_str
+    AddOutput(2, generated_nonce_hex, out_work_order_data); // generated_nonce_hex_str
+    AddOutput(3, enc_sym_key, out_work_order_data); // Encrypted symmetric key
+    AddOutput(4, enc_state_data, out_work_order_data); // Encrypted state data
+    AddOutput(5, hash_signed, out_work_order_data); // Signed hash
+    ByteArray serialized_pub_sig_key = StrToByteArray(enclave_data->get_serialized_signing_key());
+    AddOutput(6, serialized_pub_sig_key, out_work_order_data);
```

Comment:
why is sending public signing key of primary key necessary? replica KME already have primary KME worker details which should have public signing key.

Reply:
Removed. Leftover from debugging.